### PR TITLE
Added user, authentication and admin endpoints to AysncAPI

### PR DESCRIPTION
**Pull request by gazconroy**
Preview at https://playground.asyncapi.io/

### CODE REVIEW

File: reference/lemmy.v1.yaml [hunk 1]
```diff
@@ -0,0 +1,668 @@
+openapi: 3.0.0
+info:
+  title: lemmy
+  version: '1.0'
+servers:
+  - url: 'http://localhost:3000'
```
Comment (Nutomic) [inline reference/lemmy.v1.yaml:6]: You could also use one of our test servers (or multiple):

- https://voyager.lemmy.ml/
- https://enterprise.lemmy.ml/
- https://ds9.lemmy.ml/

**Comment by dessalines** [comment]
@gazconroy what's the preferred way to generate html from this file?

**Comment by gazconroy** [comment]
@dessalines . The [AsyncAPI React component](https://github.com/asyncapi/asyncapi-react) appears to be the favoured approach. For instance, the following uses its Web Component version to render the spec dynamically

```
<!DOCTYPE html>
<html>
  <head>
    <title>Lemmy WebSocket API</title>
    <!-- needed for adaptive design -->
    <meta charset="utf-8"/>
  </head>
  <body>
<script src="https://unpkg.com/@webcomponents/webcomponentsjs@2.5.0/webcomponents-bundle.js"></script>
<script src="https://unpkg.com/@asyncapi/web-component@0.19.0/lib/asyncapi-web-component.js" defer></script>
<asyncapi-component
  schemaUrl="asyncapi.yaml"
  cssImportPath="https://unpkg.com/@asyncapi/react-component@0.19.0/lib/styles/fiori.css">
</asyncapi-component>
</body>
</html>
```

**Comment by dessalines** [comment]
@gazconroy , We've just recently moved all docs and API docs to our [joinlemmy-site](https://github.com/LemmyNet/joinlemmy-site) repo. I set up your asyncapi.yaml file here: https://github.com/LemmyNet/joinlemmy-site/blob/main/static/scripts/asyncapi.yaml

You can move any changes / additions to that file over there, then once its ready we can start sharing that link in other places.

The [joinlemmy-site](https://github.com/LemmyNet/joinlemmy-site) repo has instructions on how to build that locally if you'd like to preview the html it generates from your file there. Let me know if there's anything else you need or any questions.

**Pull request closed without merging**